refactor: Quoted Item Comparison Report

marination opened the pull request:
- Refactored Report View with entries grouped by supplier and RFQ reference too.
![screencapture-ugh-in-8001-desk-2020-05-07-17_30_06](https://user-images.githubusercontent.com/25857446/81292826-ebd8e180-9089-11ea-92d2-f49f8c969c1c.png)

- Added Bar Graph and additional functional filters.
- Added Supplier Multi Select filter to compare between specific suppliers.

**Working:**
![quoted-item-comp](https://user-images.githubusercontent.com/25857446/81292985-2c385f80-908a-11ea-8b29-9c16aebcb0cd.gif)

- Added **'valid_till'** date in Supplier Quotation along with Expiry Status
![Screenshot 2020-04-15 at 11 49 53 AM](https://user-images.githubusercontent.com/25857446/79308000-76a83f80-7f15-11ea-92e8-6e4e6092c54c.png)

- Added patch to set `valid_till` and daily job to set **expiry** **status**

marination committed:
refactor: Quoted Item Comparison Report

- Refactored Report View
- Added Chart
- Added 'valid_till' date in Supplier Quotation
- Added patch to set valid_till and daily job to set expiry status

marination committed:
Merge branch 'develop' of https://github.com/frappe/erpnext into quoted-item-report

marination committed:
fix: Patch fix, Travis fix and cleanup
- Added UOM column in Report
- Removed mandatory on `valid_till`
- Added list view indicator for Expired status in Supplier Quotation
- Sorted Labels in Chart and syntax cleanup
- Made labels Translatable
- Fixed patch

rohitwaghchaure commented on erpnext/buying/report/quoted_item_comparison/quoted_item_comparison.js:
```diff
@@ -37,8 +35,83 @@ frappe.query_reports["Quoted Item Comparison"] = {
+		}
+
+		// create  datapoints for each qty
+		for (let supplier of Object.keys(supplier_wise_map)) {
```
```
for (let supplier in supplier_wise_map) {
}
```

rohitwaghchaure commented on erpnext/buying/report/quoted_item_comparison/quoted_item_comparison.py:
```diff
@@ -2,103 +2,123 @@
+			`tabSupplier Quotation Item` sqi,
+			`tabSupplier Quotation` sq
+		WHERE
+			sqi.item_code = '{0}'
```
If item has special character then this may break, either do frappe.db.escape(item) or %s (see below query) 

```
	supplier_quotation_data = frappe.db.sql("""SELECT
		sqi.parent, sqi.qty, sqi.rate, sqi.uom, sqi.request_for_quotation,
		sq.supplier
		FROM
			`tabSupplier Quotation Item` sqi,
			`tabSupplier Quotation` sq
		WHERE
			sqi.item_code = %s
			AND sqi.parent = sq.name
			AND sqi.docstatus < 2
			AND sq.company = %s
			AND sq.status != 'Expired'
			{0}""".format(conditions), (item, filters.get("company")), as_dict=1)
```

marination committed:
fix: Query enhancement, cleanup, added extra filter

- Query changes as requested
- Moved chart generation from js to py
- Added Supplier Multiselect filter

marination committed:
Merge branch 'develop' into quoted-item-report

marination commented:
@rohitwaghchaure done, please review.

marination committed:
Merge branch 'develop' into quoted-item-report

rohitwaghchaure committed:
Merge branch 'develop' into quoted-item-report

rohitwaghchaure committed:
Fixed typo
